nvme-cli: Implement list command that doesn't rely on libudev

ScottyBauer opened the pull request:
The previous list command required libudev. We can accomplish the same
thing with standard functions, and in the future remove the libudev
dependency

Signed-off-by: Scott Bauer <scott.bauer@intel.com>

keithbusch commented:
Awesome, this will fix several users' concerns.

I'm thinking we can drop the libudev optional dependency entirely and have only your method so that binaries can work across different systems.

The man page in Documentation/nvme-list.txt declares the udev dependency, which we can drop with this fix.

Finally, a possibly cleaner method might use 'scandir' to collect dirents instead of implementing looping over entries on your own, but no big deal.

ScottyBauer commented:
Alrighty, 

Removed the ifdefs from nvme.c and removed the udev stuff from the documentation. There is still use of udev code in fabrics. c so we cannot remove the libudev checks in the make file yet.

Tested it on cent OS 7 vm with one device.

keithbusch commented:
This looks good to me. I'll give it a test today as well and merge if no problems.
